exec/java-exec/src/main/java/org/apache/drill/exec/ops/QueryContext.java

```diff
@@ -77,6 +78,7 @@
   private final SchemaTreeProvider schemaTreeProvider;
   /** Stores constants and their holders by type */
   private final Map<String, Map<MinorType, ValueHolder>> constantValueHolderCache;
+  private StatementType stmtType = StatementType.UNKNOWN;
```

Comment:
Passing it in the constructor seems infeasible because the query context is created much earlier. However, the statement type is determined after parsing.